File: exam-system-java/server/src/main/java/com/yyjy/exam/user/repository/UsersRepository.java
package com.yyjy.exam.user.repository;

import com.yyjy.exam.entity.user.entity.Users;
import com.yyjy.exam.entity.user.entity.UsersTable;
import org.babyfish.jimmer.spring.repository.JRepository;
import org.babyfish.jimmer.sql.fetcher.Fetcher;

import java.util.Optional;

public interface UsersRepository extends JRepository<Users, Long> {
	
	Optional<Users> findByEmail(String email);

	Optional<Users> findByUuid(String uuid);
	
	default Optional<Users> findByEmail(String email, Fetcher<Users> fetcher) {
		var userT = UsersTable.$;
		return Optional.ofNullable(sql().createQuery(userT)
				                           .where(userT.email().eq(email))
				                           .select(userT.fetch(fetcher))
				                           .fetchFirstOrNull());
	}
}

File: exam-system-server/server/src/main/kotlin/com/yyjy/repository/UsersRepository.kt
package com.yyjy.repository

import com.yyjy.models.entity.Users
import com.yyjy.models.entity.email
import org.babyfish.jimmer.spring.repository.KRepository
import org.babyfish.jimmer.sql.kt.ast.expression.eq

/**
 * @author Nyxcirea
 * date 2026/4/1
 * description: TODO
 */
interface UsersRepository : KRepository<Users, Long> {

    fun findByEmail(email: String?): Users? = sql.createQuery(Users::class) {
        where(table.email eq email)
        select(table)
    }.fetchOneOrNull()
}
